lang/luasocket/Makefile

```diff
@@ -8,33 +8,59 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=luasocket
-PKG_SOURCE_VERSION:=6d5e40c324c84d9c1453ae88e0ad5bdd0a631448
-PKG_VERSION:=3.0-rc1-20130909
-PKG_RELEASE:=5
+PKG_SOURCE_DATE:=2019-04-21
+PKG_SOURCE_VERSION:=733af884f1aa18ff469bf3c4d18810e815853211
+PKG_RELEASE:=1
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
-PKG_MIRROR_HASH:=d2fa075d8bd026c41e0eb1a634ac2ad8115dee8abb070720e8e91fab51f86ee4
-PKG_SOURCE_URL:=https://github.com/diegonehab/luasocket.git
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE_URL:=https://github.com/diegonehab/luasocket
+PKG_MIRROR_HASH:=60aef7544426cae3e6c7560a6e4ad556a04b879ca0ad0311645b2c513c872128
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(BUILD_VARIANT)/$(PKG_NAME)-$(PKG_SOURCE_DATE)-$(call version_abbrev,$(PKG_SOURCE_VERSION))
```

Comment:
why get version_abbrev involved? that's some dark internal magic, when you really jus tneed to cover the "version" + "variant" variations.  I had to go look up what version_abbrev even did. keep th edate, it's human friend, but for the build dir, who cares if it's a full hash or not? it's simpler than jumping via an indirection of internally declared rules.

Reply:
@karlp - @neheb made me remove the PKG_SOURCE_SUBDIR.  When I do this the only way for me to get the patch build step to find the build directory is when I specify the version abbreviated. 
 So after I removed the PKG_SOURCE_SUBDIR, the only way I could get the patch build step to succeed was to make the PKG_BUILD_DIR match the directory that was there.  Personally, I thought it was much easier to read the makefile when PKG_SOURCE_SUBDIR was defined, but I was told to remove it.  It's likely I'm missing something that magically makes it all work, but I don't know what it is.

Reply:
@cwluxul: I took your commit (Pull Request) and dropped `PKG_BUILD_DIR` from it. It seems to be working fine, see:
```
rmilecki@elitebook:~/openwrt> make package/feeds/packages/luasocket/prepare V=s
make[1]: Entering directory '/home/rmilecki/openwrt'
make[2]: Entering directory '/home/rmilecki/openwrt-packages/lang/luasocket'
. /home/rmilecki/openwrt/include/shell.sh; xzcat /home/rmilecki/openwrt/dl/luasocket-2019-04-21-733af884.tar.xz | tar -C /home/rmilecki/openwrt/build_dir/target-arm_cortex-a9_musl-1.1.16_eabi/luasocket-2019-04-21-733af884/.. -xf - 
[ ! -d ./src/ ] || cp -fpR ./src/. /home/rmilecki/openwrt/build_dir/target-arm_cortex-a9_musl-1.1.16_eabi/luasocket-2019-04-21-733af884

Applying ./patches/0001-Add-interface-support.patch using plaintext: 
patching file src/if.c
patching file src/if.h
patching file src/luasocket.c
patching file src/makefile
patching file src/options.c

Applying ./patches/0301-Fix-mpc85xx-build.patch using plaintext: 
patching file src/makefile

Applying ./patches/040-remove-fpic-and-warnings.patch using plaintext: 
patching file src/makefile
touch /home/rmilecki/openwrt/build_dir/target-arm_cortex-a9_musl-1.1.16_eabi/luasocket-2019-04-21-733af884/.prepared_ad0594b75b61be5214667dae78c9902168b329da9893e34099c7d8ad5cb9c940
make[2]: Leaving directory '/home/rmilecki/openwrt-packages/lang/luasocket'
make[2]: Entering directory '/home/rmilecki/openwrt-packages/lang/luasocket'
make[2]: Leaving directory '/home/rmilecki/openwrt-packages/lang/luasocket'
make[1]: Leaving directory '/home/rmilecki/openwrt'
```

Reply:
@rmilecki - I thought I had to specify that.  I've removed it.